homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -285,11 +282,28 @@ def _calculate_new_probability(self):
 
         for obs in self.current_observations.values():
             if obs is not None:
-                prior = update_probability(
-                    prior,
-                    obs["prob_given_true"],
-                    obs.get("prob_given_false", 1 - obs["prob_given_true"]),
-                )
+                if obs["observation"] is True:
+                    prior = update_probability(
+                        prior,
+                        obs["prob_given_true"],
+                        obs.get("prob_given_false", 1 - obs["prob_given_true"]),
+                    )
+                elif obs["observation"] is False:
+                    prior = update_probability(
+                        prior,
+                        1 - obs["prob_given_true"],
+                        1 - obs.get("prob_given_false", 1 - obs["prob_given_true"]),
```

Comment:
This doesn't seam right

If prob_given_false is missing then this means: probability of being false is equal to the probability of being true

```suggestion
                         obs.get("prob_given_false", 1 - obs["prob_given_true"]),
```

Reply:
you can only do `P(not a) = 1-P(a)` if **`a`** is an independent variable, if `**a**` is a conditional variable P(a|b) then you cannot do that.

So P(off| not Asleep) = 1 - P(on| not Asleep) is not correct 

to calculate P(off | not asleep) you really need to go through the Bayes Rule:

P(a|b) = P(b|a).P(a) / ( P(b|a).P(a) + P(b| not a).P(not a))

The current implementation of this sensor works only with independent variables (it's a know limitation)

Reply:
> to calculate P(off | not asleep) you really need to go through the Bayes Rule:
I'm not sure that is true if I wanted to get from P(on| not Asleep) to P(not Asleep| on) I would need bayes rule.

> So P(off| not Asleep) = 1 - P(on| not Asleep) is not correct
I think it *is* correct in all cases where P(on| not Asleep) + P(off| not Asleep) = 1, that is to say the sensor has only two states 'on' and 'off'. Said in words, given any condition (Asleep/not asleep) there can only be two states (on/off).

Reply:
Said another way
do you not believe that these two observations are equivalent - that is all we are achieving in lines 291-295
```
observations:
  - platform: state
    entity_id: sensor.tv
    prob_given_true: 0.33  #4 hours of TV per 10 hours I’m home
    prob_given_false: 0.018 #15 minutes of TV per 14 hours away, I don’t like zeros here
    to_state: on
  - platform: state
    entity_id: sensor.tv
    prob_given_true: 0.66 # 8/12 Hours off while home
    prob_given_false: 0.982 # 13.75/14 hours off while I’m away
    to_state: off # this is the “TRUE” state now
```

Reply:
https://math.stackexchange.com/q/825698

Reply:
> I think it _is_ correct in all cases where P(on| not Asleep) + P(off| not Asleep) = 1, that is to say the sensor has only two states 'on' and 'off'. Said in words, given any condition (Asleep/not asleep) there can only be two states (on/off).

This would mean that `on` is independent of `not Asleep`

Reply:
This online calculator works in the same way:

![image](https://user-images.githubusercontent.com/11440490/161303285-fb46d407-e67b-4f96-9950-19c8d06cfd85.png)

Reply:
> This would mean that `on` is independent of `not Asleep`

I doesn't require independence 

see: This post discusses exactly this question https://math.stackexchange.com/q/825698

Reply:
Right, I stand corrected.

But the discussion started around this line of code:

`1 - obs.get("prob_given_false", 1 - obs["prob_given_true"]),`

which should fill the `prob_given_false` argument of `update_probability`

Now if you don't supply `prob_given_false` you get `1 - obs["prob_given_true"]` and you end up with `prob_given_false`  being equal to `obs["prob_given_true"]`

Reply:
Yes, this is expected behaviour
https://github.com/home-assistant/core/blob/816730cdf03e28ca52342895e01b6e766561f3c3/homeassistant/components/bayesian/binary_sensor.py#L285-L296

if `prob_given_false` is undefined then `1 - prob_given_true` is used - this is a choice by the original author of the integration.  It is a slightly odd choice as P(A|B) = 1 - P(A|B') is only true in a minority of cases

> `prob_given_false` being equal to `obs["prob_given_true"]`
This is deliberate, in a negative observation the probabilities are flipped as agreed above

Reply:
this PR changes the behaviour of the sensor, so we should at least leave it working correctly (as it should have always have been) instead of postponing a new change of behaviour to a next PR (but this is IMHO)

Reply:
I think that may well be a fair point. As per the discussion, there are cases where it is true, but only by coincidence. I will create a git diff for a change to remove the optionality and we can implement if other reviewers agree.

Reply:
Moved to new review https://github.com/home-assistant/core/pull/67631#pullrequestreview-930242477